src/main/java/org/usfirst/frc4048/common/apriltags/TCPServer.java

```diff
@@ -0,0 +1,120 @@
+package org.usfirst.frc4048.common.apriltags;
+
+import edu.wpi.first.wpilibj.DriverStation;
+
+import java.io.BufferedInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.LinkedTransferQueue;
+
+public abstract class TCPServer<T> extends Thread {
+
+    private final int port;
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private DataInputStream dataInputStream;
+    private BufferedInputStream bufferedInputStream;
+    private boolean running = false;
+    private final LinkedTransferQueue<T> readings = new LinkedTransferQueue<>();
+
+    /**
+     * @param port for network communication. Port must match between client and server (range 5800-5810 (rio limitations))
+     */
+    public TCPServer(int port) {
+        this.port = port;
+    }
+
+
+    @Override
+    public synchronized void start() {
+        try {
```

Comment:
normally there is no reason to override `start` - why did you think this is necessary?

Reply:
I think my reasoning was that I did not want to throw an error when calling the constructor. Not sure if this is the best solution.

Reply:
I would say that all the code should bio to the `run()` method

Reply:
done